src/CommonITILObject.php

```diff
@@ -8270,7 +8270,12 @@ public function handleNewItemNotifications(): void
             NotificationEvent::raiseEvent('new', $this);
 
             $status = $this->fields['status'] ?? null;
-            if (in_array($status, $this->getSolvedStatusArray())) {
+            $solution = new ITILSolution();
+            $has_solution = $solution->getFromDBByCrit([
+                'itemtype' => Ticket::class,
+                'items_id' => $this->getID(),
+            ]);
+            if (in_array($status, $this->getSolvedStatusArray()) && !$has_solution) {
```

Comment:
A ticket can have multiple solutions throughout its lifecycle.
This update prevents emails from being sent when new solutions are added to a ticket, especially in cases where the initial solution was, for example, rejected.